File: docs/snippets/all/archetypes/pinhole_perspective.rs
//! Logs a point cloud and a perspective camera looking at it.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_pinhole_perspective").spawn()?;

    let fov_y = std::f32::consts::FRAC_PI_4;
    let aspect_ratio = 1.7777778;
    rec.log(
        "world/cam",
        &rerun::Pinhole::from_fov_and_aspect_ratio(fov_y, aspect_ratio)
            .with_camera_xyz(rerun::components::ViewCoordinates::RUB)
            .with_image_plane_distance(0.1)
            .with_color(rerun::Color::from_rgb(255, 128, 0))
            .with_line_width(0.003),
    )?;

    rec.log(
        "world/points",
        &rerun::Points3D::new([(0.0, 0.0, -0.5), (0.1, 0.1, -0.5), (-0.1, -0.1, -0.5)])
            .with_radii([0.025]),
    )?;

    Ok(())
}

File: docs/snippets/all/archetypes/pinhole_perspective.cpp
// Logs a point cloud and a perspective camera looking at it.

#include <rerun.hpp>

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_pinhole_perspective");
    rec.spawn().exit_on_failure();

    const float fov_y = 0.7853982f;
    const float aspect_ratio = 1.7777778f;
    rec.log(
        "world/cam",
        rerun::Pinhole::from_fov_and_aspect_ratio(fov_y, aspect_ratio)
            .with_camera_xyz(rerun::components::ViewCoordinates::RUB)
            .with_image_plane_distance(0.1f)
            .with_color(rerun::Color(255, 128, 0))
            .with_line_width(0.003f)
    );

    rec.log(
        "world/points",
        rerun::Points3D({{0.0f, 0.0f, -0.5f}, {0.1f, 0.1f, -0.5f}, {-0.1f, -0.1f, -0.5f}}
        ).with_radii({0.025f})
    );
}
